modules/simple_forms_api/app/services/simple_forms_api/s3_service/archive_submission_to_pdf.rb

```diff
@@ -0,0 +1,184 @@
+        20-10206 20-10207 21-0845 21-0966 21-0972 21-10210
+        21-4138 21-4142 21P-0847 26-4555 40-0247 40-10007
+      ].freeze
+
+      def initialize(submission_id: nil, submission: nil, **options)
+        defaults = default_options.merge(options)
+
+        @failures = []
+        @submission = submission || FormSubmission.find(submission_id)
+
+        assign_instance_variables(defaults)
+      end
+
+      def run
+        log_info("Processing submission ID: #{submission.id}")
+        process_submission_files
+        output_directory_path
+      rescue => e
+        handle_error("Failed submission: #{submission.id}", e, { submission_id: submission.id })
+      end
+
+      private
+
+      def default_options
+        {
+          file_path: nil, # file path for the PDF file to be archived
+          include_json_archive: true, # include the form data as a JSON object
+          include_text_archive: true, # include the form data as a text file
+          metadata: {},
+          parent_dir: 'vff-simple-forms',
+          quiet_pdf_failures: true, # skip PDF generation silently
+          quiet_upload_failures: true, # skip problematic uploads silently
+          run_quiet: true, # silence but record errors, logged at the end
+          uploads_path: ['uploadedFile'] # hierarchy where the attachments can be found
+        }
+      end
+
+      def process_submission_files
+        write_pdf
+        write_as_json_archive if include_json_archive
+        write_as_text_archive if include_text_archive
+        write_user_uploads if user_uploads.present?
+        write_metadata
+      end
+
+      def write_pdf
+        encoded_pdf = generate_pdf_content
+        pdf = save_file_to_s3(
+          "#{output_directory_path}/form_#{submission.form_data['form_number']}.pdf",
+          Base64.decode64(encoded_pdf)
+        )
+        sign_s3_file_url(pdf)
+      rescue => e
+        quiet_pdf_failures ? write_pdf_error(e) : raise(e)
+      end
+
+      def generate_pdf_content
+        raise 'Missing PDF file to upload' unless file_path
+
+        Faraday::UploadIO.new(file_path, Mime[:pdf].to_s, File.basename(file_path))
+      end
+
+      def sign_s3_file_url(pdf)
+        signed_url = pdf.presigned_url(:get, expires_in: 1.year.to_i)
+        submission.form_submission_attempts&.last&.update(signed_url:)
+      end
+
+      def write_pdf_error(error)
+        log_error("PDF generation failed for submission: #{submission.id}", error)
+        save_file_to_s3("#{output_directory_path}/pdf_generating_failure.txt", error_details(error))
+      end
+
+      def error_details(error)
+        "#{error.message}\n\n#{error.backtrace.join("\n")}"
+      end
+
+      def write_as_json_archive
+        save_file_to_s3("#{output_directory_path}/form_text_archive.json", JSON.pretty_generate(form_json))
+      end
+
+      def write_as_text_archive
+        save_file_to_s3("#{output_directory_path}/form_text_archive.txt", form_text_archive.to_json)
```

Comment:
Previously I think when we did this we would pull everything down, and then zip it, and then only upload 1 zip file, with the name of the file being the va.gov internal submission ID, so that you had 1 thing to pass along and knew 1 file essentially represented 1 single claim. So less work to pass/move all around, and also less chance of only passing around partial amounts of claims and such. It also eliminated partial uploads, if the zip was there, it worked in its entirety, if the zip wasn't there, something didn't work.  With this it is possible for some portions of a claim to upload, and someone think it is complete just looking from the S3 side, when in fact components of it may have failed. This is probably for @SamStuckey as I think he changed this from zip to not-zip.

Reply:
Ahh, good callout. I'll look at the original implementation again and determine the best approach to zipping up the files.

Reply:
Well, @SamStuckey moved away from that. There might have been a good reason, Im just not sure.

Reply:
Do we want to zip anything? I thought we'd want a `.pdf` file so we can supply it directly to the end user. Or are we expecting ourselves (or the user) to unzip it upon download?

Reply:
Hmm, it might depend on if there are attachments. If there are attachments, then zip but otherwise maybe just provide the pdf?